src/query/atomWithInfiniteQuery.ts

```diff
@@ -120,10 +123,12 @@ export function atomWithInfiniteQuery<
 
       const observer = new InfiniteQueryObserver(queryClient, defaultedOptions)
 
-      observer
-        .fetchOptimistic(defaultedOptions)
-        .then(listener)
-        .catch((error) => listener({ error }))
+      if (!initialData) {
+        observer
+          .fetchOptimistic(defaultedOptions)
+          .then(listener)
+          .catch((error) => listener({ error }))
```

Comment:
@dai-shi I don't think we need this bit of code either way it seems -- can you tell me more about it?

Because we are subscribing on mount, the subscription kicks off the fetch initially. Seems like we're doing double work, unless I am missing something

Reply:
When an atom suspends, the component suspends and it will not be mounted, which means `onMount` will not be called. Unless we resolve the promise, the component will not be mounted.